schema.sql

```diff
@@ -47,3 +47,29 @@ REFERENCES species(id);
 ALTER TABLE animals ADD CONSTRAINT fk_owner_id
 FOREIGN KEY (owner_id)
 REFERENCES owners(id);
+
+--  species table 
+
+DROP TABLE IF EXISTS species;
+
+CREATE TABLE species (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    name CHAR(100),
+    PRIMARY KEY (id)
+);
+
+-- Create the specializations table
+CREATE TABLE specializations (
+    vet_id INT REFERENCES vets(id),
+    species_id INT REFERENCES species(id),
+    PRIMARY KEY (vet_id, species_id)
+);
+
+
+-- Create the visits table
+CREATE TABLE visits (
+    animal_id INT REFERENCES animals(id),
+    vet_id INT REFERENCES vets(id),
+    visit_date DATE,
+    PRIMARY KEY (animal_id, vet_id, visit_date)
+);
```

Comment:
- [ ] As required for the project, kindly ensure that you have created a table named `vets` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
  - `age`: integer
  - `date_of_graduation`: date

- [ ] Once you have created the `vets` table you should ensure thar:
  * There is a **many-to-many** relationship between the tables `species` and `vets`: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.
  * There is a **many-to-many** relationship between the tables `animals` and `vets`: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.